File: all-submissions/0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int currArea = -1;
        int maxArea = currArea;
        int i = 0;
        int j = height.length-1;

        while (i < j) {
            currArea = (j-i) * Math.min(height[i], height[j]);
            maxArea = Math.max(maxArea, currArea);
            if (height[i] < height[j]) {
                i++;
            }
            else {
                j--;
            }
        }
        return maxArea;

        // for(int i=0; i<height.length-1; i++){
        //     for (int j = i+1 ; j<height.length; j++) {
        //         currArea = (j-i) * Math.min(height[i], height[j]);
        //         if (currArea > maxArea) {
        //             maxArea = currArea;
        //         }
        //     }
        // }
        // return maxArea;
        
    }
}

File: all-submissions/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        if height is None or len(height) == 1:
            return 0
        left = 0 
        right = len(height)-1
        maxWater = 0
        while(left<right):
            h = min(height[left], height[right])
            w = right - left
            currMax = h*w
            maxWater = max(maxWater, currMax)
            if height[left] < height[right]:
                left +=1
            else:
                right -=1
        return maxWater	


